README.md

```diff
@@ -2,12 +2,30 @@
 
-The full API documentation for this package is automatically generated from the `master` via [typedoc](https://typedoc.org) branch and published to https://node-resque.actionherojs.com/
+The full API documentation for this package is automatically generated from the `main` via [typedoc](https://typedoc.org) branch and published to https://node-resque.actionherojs.com/
 
 [![Nodei stats](https://nodei.co/npm/node-resque.png?downloads=true)](https://npmjs.org/package/node-resque)
 
 [![CircleCI](https://circleci.com/gh/actionhero/node-resque.svg?style=svg)](https://circleci.com/gh/actionhero/node-resque)
 
+## The Resque Factory (How It Works)
+
+### Overview
+
+Resque is a queue based task processing system that can be thought of as a "Kanban" style factory. Workers in this factory can each only work one Job at a time. They pull Jobs from Queues and work them to completion (or failure). Each Job has two parts: instructions on how to complete the job (the `perform` function), and any inputs necessary to complete the Job.
+
+### Queues
+
+In our factory example, Queues are analogous to conveyor belts. Jobs are placed on the belts (Queues) and are held in order waiting for a Worker to pick them up. There are three types of Queues: regular work Queues, Delayed Job Queues, and the Failed Job Queue. The Delayed Job Queues contains Job definitions that are intended to be worked at or in a specified time. The Failed Job Queue is where Workers place any Jobs that have failed during execution.
+
+### Workers
+
+Our Workers are the heart of the factory. Each Worker is assigned one or more Queues to check for work. After taking a Job from a Queue the Worker attempts to complete the Job. If successful, they go back to check out more work from the Queues. However, if there is a failure, the Worker records the job and its inputs in the Failed Jobs Queue before going back for more work.
+
+### Scheduler
+
+The Scheduler can be thought of as a specialized type of Worker. Unlike other Workers, the Scheduler does not execute any Jobs, instead it manages the Delayed Job Queue. As Job definitions are added to the Delayed Job Queue they must specify when they can become available for execution. The Scheduler constantly checks to see if any Delayed Jobs are ready to execute. When a Delayed Job becomes ready for execution the Scheduler places a new instance of that Job in its defined Queue.
```

Comment:
Ok! fine, it's like a worker.

In the default configuration of a Scheduler, timeout is configure at 5000 ms.

Timeout is the loop time ?

What happened if the timeout scheduler check only every 5 minutes by exemple? Or even worst, 1 hour? 
Is there any chance to loose a clock time? Or it will be meet for sure?

Reply:
More specifically, that timeout is the idle time between the last Delay Queue being checked for jobs and the first Delay Queue being checked. If you have three delayed jobs (every 10 seconds, every 30 seconds, and every 60 seconds), your Delay Queues would look like [delay_10000, delay_30000, delay_60000] and the polling behavior of the Scheduler would look something like `Check delay_10000, check delay_30000, check delay_60000, sleep 5000, check delay_10000, ...`

Reply:
I have few worker working like a CRON.

- 1x Every minutes
- 4x scheduler every 5 minutes
- 1x scheduler every 15 minutes
- 1x scheduler every day

Do you see any problem on my configuration? or there is no problem

Reply:
I don't see a problem with that.